### ISSUE TITLE
[Android][Web] Layer switching does not occur upon hitting backspace

### ISSUE THREAD

**Issue by eddieantonio** [keymanapp/keyman#2349]
**Describe the bug**

tl;dr: the `layer()` command in kmn files does not switch to a non-default layer?

Say I have a KMN that has these lines:

```kmn
store(kwV) 'ᑵᑷᑹᑻᑽᑿᒁ'

begin Unicode > use(main)
group(main) using keys
  any(kwV) + [K_BKSP] > U+1420 U+1424 layer('kwV')
```

This is intended to break down any of ᑵᑷᑹᑻᑽᑿᒁ into ᐠᐤ and switch to the `kwV` later upon hitting backspace

This is what my `kwV` layer looks like:

![kwv-layer](https://user-images.githubusercontent.com/2294397/68976531-30f31980-07b3-11ea-9fd4-eac8989561a0.jpg)

When I try this in Keyman for Android, I get this:

![incorrect-layer-switching](https://user-images.githubusercontent.com/2294397/68976617-6bf54d00-07b3-11ea-981a-8cfa7a77b6d4.gif)

It started in the `default` layer, but it didn't switch to the `kwV` layer, as I expected from the declaration in the .kmn file. Instead, it remained in the `default` layer.

**Expected behavior**

Using this keyboard, when I hit backspace after any of the following: ᑵᑷᑹᑻᑽᑿᒁ, it should replace the character with ᐠᐤ and switch unconditionally to the `kwV` layer; thus it should look like this:

![Screenshot_20191115-141127](https://user-images.githubusercontent.com/2294397/68976874-14a3ac80-07b4-11ea-8714-ae447bc3cbf3.jpg)

---

(Delete the irrelevant sections)

  
**iPhone/iPad/Android (please complete the following information):**
 - Device: OnePlus 3T, Samsung Galaxy A10
 - OS: Android 9.x
 - Keyman version 12.0.4207
 - Target application: Google Notes, Keyman

---
 
**Keyboard**
 - Keyboard name: nrc_crk_cans
 - Keyboard version: 1.0.0-rc.2
 - Language name: Plains Cree (Syllabics)

Get it here:

[nrc_crk_cans.kmp](https://github.com/eddieantonio/plains-cree-touch-keyboard/releases/download/v1.0.0-rc.2/nrc_crk_cans.kmp)
[nrc_crk_cans.kmn](https://github.com/eddieantonio/plains-cree-touch-keyboard/releases/download/v1.0.0-rc.2/nrc_crk_cans.kmn)
[nrc_crk_cans.keyman-touch-layout](https://github.com/eddieantonio/plains-cree-touch-keyboard/releases/download/v1.0.0-rc.2/nrc_crk_cans.keyman-touch-layout)

**Comment by mcdurdin** [comment]
Thanks for the great report. Joshua will investigate soon.

**Comment by jahorton** [comment]
This is (at least partially) a keyboard coding/design error.  From the .touch_layout file:

```
              {
                "id": "K_BKSP",
                "text": "*BkSp*",
                "sp": "1",
                "nextlayer": "default"
              }
```

This block appears quite a number of times, _including for the default layer_.  Interestingly, this does _not_ appear in the numeric layer.  The BKSP from the numeric layer still works properly as a result.

Since K_BKSP has a defined "next layer" property, it automatically triggers **_after_** all keystroke/rule processing is complete... thus **_after_** any rules that would also result in layer changes.

We could consider adding logic for an "override" to avoid this scenario - we'd just need to have a single-keystroke-level flag that tracks if a matched rule triggered a layer change.  This doesn't _yet_ exist in KMW.

**Comment by mcdurdin** [comment]
> We could consider adding logic for an "override" to avoid this scenario - we'd just need to have a single-keystroke-level flag that tracks if a matched rule triggered a layer change. This doesn't _yet_ exist in KMW, but would probably be a reasonable feature to implement.

Seems to me we have a question of precedence. I think the `nextlayer` flag in the touch layout should take precedence as it is applied after the rule. In any case, the simple answer is to remove the `nextlayer` flag from the backspace key and apply layer switches purely in the .kmn source in this instance?

**Comment by eddieantonio** [comment]
> This is (at least partially) a keyboard coding/design error. From the .touch_layout file:
> 
> ```
>               {
>                 "id": "K_BKSP",
>                 "text": "*BkSp*",
>                 "sp": "1",
>                 "nextlayer": "default"
>               }
> ```
> 
> This block appears quite a number of times, _including for the default layer_. Interestingly, this does _not_ appear in the numeric layer. The BKSP from the numeric layer still works properly as a result.

Ugh, this is a copy-paste error :weary: I'll make an update to the keyboard to remove the `nextlayer` from the backspace keys.

About touch-layout vs. kmn precedence, I think the existing behavior is fine (touch-layout overrides kmn), however, this must be documented somewhere. So let's call it a documentation bug :sweat_drops:
